<?xml version="1.0" ?>
<launch>
  <arg name="config"	default="aist"/>
  <arg name="scene"	default="$(arg config)_base_scene"/>
  <arg name="sim"	default="false"/>

  <param name="use_real_robot"	value="$(eval not arg('sim'))"/>

  <!-- Launch real robots if "sim" == "false" -->
  <include file="$(find o2as_examples)/launch/connect_real_robots.launch"
	   if="$(eval arg('config') == 'o2as' and not arg('sim'))"/>
  <include file="$(find aist_examples)/launch/connect_$(arg config)_robots.launch"
	   if="$(eval not arg('config') == 'o2as' and not arg('sim'))"/>

  <!-- Launch planner -->
  <include file="$(find o2as_moveit_config)/launch/o2as_moveit_planning_execution.launch"
  	   if="$(eval arg('config') == 'o2as')">
    <arg name="sim" value="$(arg sim)"/>
  </include>
  <include file="$(find aist_moveit_config)/launch/aist_moveit_planning_execution.launch"
	   unless="$(eval arg('config') == 'o2as')">
    <arg name="config" value="$(arg config)"/>
    <arg name="scene"  value="$(arg scene)"/>
    <arg name="sim"    value="$(arg sim)"/>
  </include>

  <!-- Launch skill server -->
  <include file="$(find o2as_skills)/launch/o2as_skill_server.launch"
	   if="$(eval arg('config') == 'o2as')"/>
  <include file="$(find aist_skills)/launch/aist_skill_server.launch"
	   unless="$(eval arg('config') == 'o2as')"/>
</launch>
